packages/dso-toolkit/src/components/action-list/action-list.scss

```diff
@@ -16,11 +16,18 @@ dso-action-list {
     &::after {
       content: "";
       background-color: colors.$bosgroen;
-      height: calc(100% - 48px);
-      left: -50px;
+      height: calc(100% - #{units.$u6});
       position: absolute;
-      top: 64px;
-      width: 4px;
+      top: #{units.$u8};
+      width: #{units.$u1 * 0.5};
+
+      @media screen and (max-width: media-query-breakpoints.$screen-xs-max) {
+        left: #{units.$u3 - units.$u1 * 0.25};
```

Comment:
Dit was niet helemaal wat ik voor ogen had toen ik het had over het herleiden van magic numbers. Je hebt er nu _magic numbers in disguise_ van gemaakt.

Mijn probleem is als ik dit lees, ik geen idee heb wat hier staat. Mijn gedachtes gaan als volgt:

* "u3 - u1 * 0.25, dat is een complex sommetje op een onhandige plek"
* "Wat is u3 ook alweer?"
* "Hmm, - en * -> rekenregels?"
* Een kwart van 8.. Waar komt 2 vandaan?
* 24 min 2 ?

Je had ook `$u11 * 0.25` kunnen doen, dat is ook 22px en laat ook niet zien waarom dit element 22px moet opschuiven.

De 22px heeft te maken met de breedte van het element `div.dso-step-counter` (48px) en de breedte van de border (4px): `48 * 0.5 - 4 * 0.5`. Of je dit in een variabele stopt kan je laten afhangen van of deze waarde nog elders wordt gebruikt, of als de uitkomst nog steeds te complex is om meteen te begrijpen en dat je nog duidelijkheid kan toevoegen met een variabele naam.

Afgezien van het verhaal wat deze formule vertelt, is het ook belangrijk voor het gebruik van dezelfde units. Want dat is nu niet het geval. De breedte van `div.dso-step-counter` is geen 48px maar `$u1 + 2rem + $u1` (border-width + width + border-width). Een gebruiker die de root font-size heeft opgehoogd ziet dit:

<img width="250" alt="image" src="https://github.com/dso-toolkit/dso-toolkit/assets/9696261/9cd16a19-ac96-4719-8cbd-dc6f8c0ca95a">

Dan zijn er nog twee specificity issues:

<img width="328" alt="image" src="https://github.com/dso-toolkit/dso-toolkit/assets/9696261/89a2a42c-d3f9-4416-a10e-a369f2285d04">

* 22px en 30px overlappen met elkaar, vanwege de conflicterende media queries.
* De breedte van de divider/separator line raakt ook de flow line. Dit is ongewenst omdat het niets met de cascade te maken heeft.

En dan nog het volgende: De interpolations zijn niet consequent:
* Sommige property values worden expliciet geinterpoleerd, anderen niet (`margin-inline-start: units.$u1;` vs `top: #{units.$u8};`).
* Voor `calc()` hetzelfde: `height: calc(100% - #{units.$u6});` vs `width: calc(100% - units.$u10);`.

De voorkeur heeft om zo min mogelijk expliciet te interpoleren. Dus de property values zijn een nobrainer.

Maar bij `calc()` wil ik nog even stilstaan, want ik begreep eigenlijk niet waarom je `calc(100% - units.$u10);` gebruikt. Ik was er even heilig van overtuigd dat er gewoon `$u10` in de CSS stond, maar dat (gelukkig) niet het geval. Na wat speurwerk ben ik erachter gekomen dat dit pas sinds Dart Sass 1.40.0 kan, voorheen moesten we expliciet interpoleren. De codebase staat daar dan ook vol van.

Maar nu het kan, laten we het absoluut gebruiken. Dus ook in `calc()` niet meer expliciet interpoleren.

> * Sass variables and functions can now be used in calc() expressions.

_https://github.com/sass/dart-sass/releases/tag/1.40.0_

Waarom willen we zo min mogelijk expliciet interpoleren? Dat heeft te maken met het type wat uit een expliciete interpolatie komt. Dit is namelijk altijd van het type "unquoted strings", en dan kunnen we er niet meer mee rekenen. Zie ook https://sass-lang.com/documentation/interpolation/ het kopje "Heads up!".

We moeten er een gewoonte van maken om juist de extra stap te zetten voor expliciet interpoleren ipv. "maar gewoon doen. Door dit consequent te doen (dus ook voor property values) verkleinen we het risico op een variable met unquoted strings.

Lang verhaal kort:
- [ ] Sass variables herleiden.
- [ ] Specificity oplossen.
- [ ] Consquent interpoleren.

@daverhelst Vink de items af als je ze hebt afgerond, dan review ik daarna nog een keer.

Reply:
@tfrijsewijk wat een verhaal, maar ik trap heel even kort op de rem. Het issue ging over het aanpassen van de smalle viewport, waarbij een aantal waarden aangepast zouden moeten worden. Dat is gedaan, en tevens heb ik de bestaande "magic numbers" inderdaad omgeschreven naar "magic numbers uit variabelen zodat het op het grid past". Maar nu wil je graag (en daar ben ik het wel mee eens) dat deze magic numbers, die al die tijd al magic waren, af te leiden zijn uit wat deze waarden nou uiteindelijk zijn. Maar gaat dat niet ver buiten de scope van dit issue?
Willen we hier niet gewoon beter een nieuw issue voor aanmaken, zodat deze change wel door kan en @MYpenburg hier gebruik van kan maken?

Reply:
Wat mij betreft zijn de punten die ik aandraag niet buiten scope, maar juist oorzaak van de changes.

Overigens gaat mijn verhaal niet alleen over de magic numbers.

@MYpenburg kan hier voorlopig geen gebruik van maken ivm update cyclus van het portaal. Er is dus geen tijdsdruk.